src/common/libflux/future.h

```diff
@@ -0,0 +1,46 @@
+#ifndef _FLUX_CORE_FUTURE_H
+#define _FLUX_CORE_FUTURE_H
+
+#include "reactor.h"
+#include "types.h"
+#include "handle.h"
+#include "dispatch.h"
+
+#include <stdbool.h>
+
+typedef struct flux_future_struct flux_future_t;
+
+typedef void (*flux_continuation_f)(flux_future_t *f, void *arg);
+
+flux_future_t *flux_future_create (flux_reactor_t *r);
+void flux_future_destroy (flux_future_t *f);
+
+int flux_future_set_timeout (flux_future_t *f, double timeout);
```

Comment:
Not if covered by man page IMHO.   I prefer not to keep documentation up to date in both places.